src/report/generate_https_scan_report.py

```diff
@@ -0,0 +1,926 @@
+        result["hsts_preloaded_percentage"] = round(
+            self.__hsts_preloaded_count / self.__domain_count * 100.0, 1
+        )
+        result["hsts_entire_domain_percentage"] = round(
+            self.__hsts_entire_domain_count / self.__domain_count * 100.0, 1
+        )
+        # result['strictly_forces_percentage'] = 0
+        # result['downgrades_percentage'] = 0
+        # result['hsts_preloaded_percentage'] = 0
+        # result['hsts_entire_domain_percentage'] = 0
+        result[
+            "domain_has_no_weak_crypto_count"
+        ] = self.__domain_has_no_weak_crypto_count
+        result["has_no_weak_crypto_percentage"] = self.__has_no_weak_crypto_percentage
+        result["bod_1801_percentage"] = self.__bod_1801_percentage
+        result["bod_1801_count"] = self.__bod_1801_count
+        result[
+            "domain_supports_https_count"
+        ] = self.__domain_supports_https_count  # added
+        result["uses_https_percentage"] = self.__uses_https_percentage
+        result["enforces_https_percentage"] = self.__enforces_https_percentage
+        result["strictly_forces_count"] = self.__strictly_forces_count
+        result["domain_enforces_https_count"] = self.__domain_enforces_https_count
+        result["hsts_count"] = self.__hsts_count
+        result["hsts_preloaded_count"] = self.__hsts_preloaded_count
+        result["hsts_preload_ready_count"] = self.__hsts_preload_ready_count
+        result["domain_uses_strong_hsts_count"] = self.__domain_uses_strong_hsts_count
+        result["https_expired_cert_count"] = self.__https_expired_cert_count
+        result["https_bad_hostname_count"] = self.__https_bad_hostname_count
+        result["https_bad_chain_count"] = self.__https_bad_chain_count
+        result["hsts_low_max_age_count"] = self.__hsts_low_max_age_count
+
+        self.__latex_escape_structure(result["report_doc"])
+
+        with open(filename, "w") as out:
+            out.write(json.dumps(result))
+
+    def __generate_latex(self, mustache_file, json_file, latex_file):
+        template = codecs.open(mustache_file, "r", encoding="utf-8").read()
+
+        with codecs.open(json_file, "r", encoding="utf-8") as data_file:
+            data = json.load(data_file)
+
+        r = chevron.render(template, data)
+        with codecs.open(latex_file, "w", encoding="utf-8") as output:
+            output.write(r)
+
+    def __generate_final_pdf(self):
+        xelatex = ["/usr/bin/xelatex", REPORT_TEX]
+        # Bandit frowns upon the use of subprocess, but we need it
+        # here.  Hence the nosec.
+        subprocess.run(xelatex)  # nosec B603
+        subprocess.run(xelatex)  # nosec B603
+
+
+def main():
+    """Create PSHTT Agency Report PDF."""
+    args = docopt(__doc__, version="v0.0.1")
+    db = db_from_config(DB_CONFIG_FILE)
+
+    print("Generating HTTPS Report for {}...".format(args['"AGENCY"']))
+    generator = ReportGenerator(db, args['"AGENCY"'], debug=args["--debug"])
```

Comment:
This updates for my comment about the Docopt specification. I'm not sure there are any other references.

```suggestion
    print(f"Generating HTTPS Report for {args['AGENCY']}...")
    generator = ReportGenerator(db, args["AGENCY"], debug=args["--debug"])
```